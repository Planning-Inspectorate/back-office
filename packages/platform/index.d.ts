import './config/dotenv';
import './http/ttl';
import './util/CryptoUtils';
import './util/date';
import './util/filter';
import './util/url';
import './validators/date';
import './validators/postcode';

declare module '@pins/platform' {
	export * from './config/dotenv';
	export * from './http/ttl';
	export * from './util/CryptoUtils';
	export * from './util/date';
	export * from './util/filter';
	export * from './util/url';
	export * from './validators/date';
	export * from './validators/postcode';
}
