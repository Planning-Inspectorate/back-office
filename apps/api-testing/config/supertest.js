// @ts-nocheck
import supertest from 'supertest';
import config from './config.cjs';
import { Logger } from '../utils/logger.js';

const logger = new Logger();
logger.log('green', '************* Tests starting ****************');
logger.log('cyan', `Running tests against: ${config.baseUrl}`);
logger.log('green', '*********************************************');

const request = supertest(config.baseUrl);

export default request;
