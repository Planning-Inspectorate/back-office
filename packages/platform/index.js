export * from './http/hooks.js';
export * from './testing/assets/assets.js';
export * from './testing/fake.js';
export * from './testing/html-parser.js';
export * from './utils/date.js';
export * from './utils/dotenv.js';
export * from './utils/filter.js';
export * from './validators/date.js';
export * from './validators/postcode.js';
