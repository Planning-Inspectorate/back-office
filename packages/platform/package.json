{
  "name": "@pins/platform",
  "version": "0.0.0",
  "type": "module",
  "main": "index.js",
  "private": true,
  "engines": {
    "node": ">=16.14.0"
  },
  "scripts": {
    "clean": "rimraf .turbo node_modules dist",
    "lint:js": "npx eslint .",
    "lint:js:fix": "npx eslint . --fix",
    "tscheck": "npx tsc -p jsconfig.json --maxNodeModuleJsDepth 0"
  },
  "dependencies": {
    "date-fns": "2.28.0",
    "dotenv": "16.0.0",
    "got": "^12.5.2",
    "html": "^1.0.0",
    "lodash-es": "^4.17.21",
    "node-html-parser": "^5.3.3",
    "randexp": "^0.5.3"
  },
  "devDependencies": {
    "@types/html": "^1.0.1",
    "@types/lodash-es": "^4.17.6",
    "type-fest": "^2.12.2"
  }
}
