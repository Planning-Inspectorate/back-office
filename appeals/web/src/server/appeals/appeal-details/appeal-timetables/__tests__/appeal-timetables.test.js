import { parseHtml } from '@pins/platform';
import nock from 'nock';
import supertest from 'supertest';
import { appealData as baseAppealData } from '#testing/app/fixtures/referencedata.js';
import { createTestEnvironment } from '#testing/index.js';

const { app, teardown } = createTestEnvironment();
const request = supertest(app);
const baseUrl = '/appeals-service/appeal-details/1/appeal-timetables';

const finalCommentReviewData = {
	...baseAppealData,
	appealTimetable: {
		appealTimetableId: 1,
		finalCommentReviewDate: '2023-08-09'
	}
};

describe('Appeal Timetables', () => {
	afterEach(teardown);

	it(`should render "Schedule Final Comment Review Date" page`, async () => {
		const appealData = {
			...baseAppealData,
			appealTimetable: {
				appealTimetableId: 1
			}
		};

		nock('http://test/').get('/appeals/1').reply(200, appealData);

		const response = await request.get(`${baseUrl}/final-comment-review`);
		const element = parseHtml(response.text);

		expect(element.innerHTML).toMatchSnapshot();
	});

	it(`should render "Change Final Comment Review Date" page`, async () => {
		nock('http://test/').get('/appeals/1').reply(200, finalCommentReviewData);

		const response = await request.get(`${baseUrl}/final-comment-review`);
		const element = parseHtml(response.text);

		expect(element.innerHTML).toMatchSnapshot();
	});

	it('should render "Change Final Comment Review Date" with error (no answer provided)', async () => {
		nock('http://test/').get('/appeals/1').reply(200, finalCommentReviewData);

		const response = await request.post(`${baseUrl}/final-comment-review`).send({
			'due-date-day': '',
			'due-date-month': '',
			'due-date-year': ''
		});
		const element = parseHtml(response.text);

		expect(element.innerHTML).toMatchSnapshot();
	});

	it('should render "Change Final Comment Review Date" with error (api error)', async () => {
		nock('http://test/').get('/appeals/1').reply(200, finalCommentReviewData);
		nock('http://test/')
			.patch('/appeals/1/appeal-timetables/1')
			.reply(400, {
				errors: {
					lpaQuestionnaireDueDate: 'must be a business day'
				}
			});

		const response = await request.post(`${baseUrl}/final-comment-review`).send({
			'due-date-day': '1',
			'due-date-month': '1',
			'due-date-year': '2050'
		});
		const element = parseHtml(response.text);

		expect(element.innerHTML).toMatchSnapshot();
	});

	it('should render "Change Final Comment Review Date" confirmation page', async () => {
		nock('http://test/').get('/appeals/1').reply(200, finalCommentReviewData);
		nock('http://test/').patch('/appeals/1/appeal-timetables/1').reply(200, {
			finalCommentReviewDate: '2050-01-02T01:00:00.000Z'
		});

		const response = await request.post(`${baseUrl}/final-comment-review`).send({
			'due-date-day': '2',
			'due-date-month': '1',
			'due-date-year': '2050'
		});

		expect(response.statusCode).toBe(302);
	});

	it(`should render "Schedule issue determination" page`, async () => {
		const appealData = {
			...baseAppealData,
			appealTimetable: {
				appealTimetableId: 1
			}
		};

		nock('http://test/').get('/appeals/1').reply(200, appealData);

		const response = await request.get(`${baseUrl}/issue-determination`);
		const element = parseHtml(response.text);

		expect(element.innerHTML).toMatchSnapshot();
	});

	it(`should render "Change issue determination" page`, async () => {
		const appealData = {
			...baseAppealData,
			appealTimetable: {
				appealTimetableId: 1,
				issueDeterminationDate: '2023-08-09'
			}
		};

		nock('http://test/').get('/appeals/1').reply(200, appealData);

		const response = await request.get(`${baseUrl}/issue-determination`);
		const element = parseHtml(response.text);

		expect(element.innerHTML).toMatchSnapshot();
	});

	it(`should render "Schedule LPA questionnaire Date" page`, async () => {
		const appealData = {
			...baseAppealData,
			appealTimetable: {
				appealTimetableId: 1
			}
		};

		nock('http://test/').get('/appeals/1').reply(200, appealData);

		const response = await request.get(`${baseUrl}/lpa-questionnaire`);
		const element = parseHtml(response.text);

		expect(element.innerHTML).toMatchSnapshot();
	});

	it(`should render "Change LPA questionnaire Date" page`, async () => {
		const appealData = {
			...baseAppealData,
			appealTimetable: {
				appealTimetableId: 1,
				lpaQuestionnaireDueDate: '2023-08-09'
			}
		};

		nock('http://test/').get('/appeals/1').reply(200, appealData);

		const response = await request.get(`${baseUrl}/lpa-questionnaire`);
		const element = parseHtml(response.text);

		expect(element.innerHTML).toMatchSnapshot();
	});

	it(`should render "Schedule statement review Date" page`, async () => {
		const appealData = {
			...baseAppealData,
			appealTimetable: {
				appealTimetableId: 1
			}
		};

		nock('http://test/').get('/appeals/1').reply(200, appealData);

		const response = await request.get(`${baseUrl}/statement-review`);
		const element = parseHtml(response.text);

		expect(element.innerHTML).toMatchSnapshot();
	});

	it(`should render "Change statement review Date" page`, async () => {
		const appealData = {
			...baseAppealData,
			appealTimetable: {
				appealTimetableId: 1,
				statementReviewDate: '2023-08-09'
			}
		};

		nock('http://test/').get('/appeals/1').reply(200, appealData);

		const response = await request.get(`${baseUrl}/statement-review`);
		const element = parseHtml(response.text);

		expect(element.innerHTML).toMatchSnapshot();
	});
});
