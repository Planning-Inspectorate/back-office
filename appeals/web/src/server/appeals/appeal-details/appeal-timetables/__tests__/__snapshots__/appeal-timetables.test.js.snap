// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appeal Timetables should render "Change Final Comment Review Date" page 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-full"><span class="govuk-caption-xl govuk-!-margin-bottom-1">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-!-margin-bottom-7">
        <h1 class="govuk-fieldset__legend--xl govuk-!-margin-top-0">Change final comment review due date</h1>
        <div class="govuk-inset-text">The current due date for the final comment review is 9 August 2023</div>
    </div>
    <form method="post" novalidate="novalidate">
        <div class="govuk-form-group">
            <div id="due-date-hint" class="govuk-hint">For example, 27 3 2007</div>
            <div class="govuk-date-input" id="due-date">
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-day">Day</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-day" name="due-date-day" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-month">Month</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-month" name="due-date-month" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-year">Year</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-4"
                        id="due-date-year" name="due-date-year" type="text" inputmode="numeric">
                    </div>
                </div>
            </div>
        </div>
        <div class="govuk-button-group">
            <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
        </div>
    </form>
</main>"
`;

exports[`Appeal Timetables should render "Change Final Comment Review Date" with error (api error) 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-error-summary" data-module="govuk-error-summary">
                <div role="alert">
                    <h2 class="govuk-error-summary__title"> There is a problem</h2>
                    <div class="govuk-error-summary__body">
                        <ul class="govuk-list govuk-error-summary__list">
                            <li><a href="#due-date-day">Date must be a business day</a>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
        </div>
    </div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-full"><span class="govuk-caption-xl govuk-!-margin-bottom-1">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-!-margin-bottom-7">
        <h1 class="govuk-fieldset__legend--xl govuk-!-margin-top-0">Change final comment review due date</h1>
        <div class="govuk-inset-text">The current due date for the final comment review is 9 August 2023</div>
    </div>
    <form method="post" novalidate="novalidate">
        <div class="govuk-form-group">
            <div id="due-date-hint" class="govuk-hint">For example, 27 3 2007</div>
            <div class="govuk-date-input" id="due-date">
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-day">Day</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-day" name="due-date-day" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-month">Month</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-month" name="due-date-month" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-year">Year</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-4"
                        id="due-date-year" name="due-date-year" type="text" inputmode="numeric">
                    </div>
                </div>
            </div>
        </div>
        <div class="govuk-button-group">
            <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
        </div>
    </form>
</main>"
`;

exports[`Appeal Timetables should render "Change Final Comment Review Date" with error (no answer provided) 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-two-thirds">
            <div class="govuk-error-summary" data-module="govuk-error-summary">
                <div role="alert">
                    <h2 class="govuk-error-summary__title"> There is a problem</h2>
                    <div class="govuk-error-summary__body">
                        <ul class="govuk-list govuk-error-summary__list">
                            <li><a href="#due-date-day">Date day cannot be empty</a>
                            </li>
                            <li><a href="#due-date-month">Date month cannot be empty</a>
                            </li>
                            <li><a href="#due-date-year">Date year cannot be empty</a>
                            </li>
                            <li><a href="#">Date must be a valid date</a>
                            </li>
                        </ul>
                    </div>
                </div>
            </div>
        </div>
    </div>
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-full"><span class="govuk-caption-xl govuk-!-margin-bottom-1">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-!-margin-bottom-7">
        <h1 class="govuk-fieldset__legend--xl govuk-!-margin-top-0">Change final comment review due date</h1>
        <div class="govuk-inset-text">The current due date for the final comment review is 9 August 2023</div>
    </div>
    <form method="post" novalidate="novalidate">
        <div class="govuk-form-group">
            <div id="due-date-hint" class="govuk-hint">For example, 27 3 2007</div>
            <div class="govuk-date-input" id="due-date">
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-day">Day</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-day" name="due-date-day" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-month">Month</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-month" name="due-date-month" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-year">Year</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-4"
                        id="due-date-year" name="due-date-year" type="text" inputmode="numeric">
                    </div>
                </div>
            </div>
        </div>
        <div class="govuk-button-group">
            <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
        </div>
    </form>
</main>"
`;

exports[`Appeal Timetables should render "Change LPA questionnaire Date" page 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-full"><span class="govuk-caption-xl govuk-!-margin-bottom-1">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-!-margin-bottom-7">
        <h1 class="govuk-fieldset__legend--xl govuk-!-margin-top-0">Change LPA questionnaire due date</h1>
        <div class="govuk-inset-text">The current due date for the LPA questionnaire is 9 August 2023</div>
    </div>
    <form method="post" novalidate="novalidate">
        <div class="govuk-form-group">
            <div id="due-date-hint" class="govuk-hint">For example, 27 3 2007</div>
            <div class="govuk-date-input" id="due-date">
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-day">Day</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-day" name="due-date-day" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-month">Month</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-month" name="due-date-month" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-year">Year</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-4"
                        id="due-date-year" name="due-date-year" type="text" inputmode="numeric">
                    </div>
                </div>
            </div>
        </div>
        <div class="govuk-button-group">
            <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
        </div>
    </form>
</main>"
`;

exports[`Appeal Timetables should render "Change issue determination" page 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-full"><span class="govuk-caption-xl govuk-!-margin-bottom-1">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-!-margin-bottom-7">
        <h1 class="govuk-fieldset__legend--xl govuk-!-margin-top-0">Change issue determination due date</h1>
        <div class="govuk-inset-text">The current due date for the issue determination is 9 August 2023</div>
    </div>
    <form method="post" novalidate="novalidate">
        <div class="govuk-form-group">
            <div id="due-date-hint" class="govuk-hint">For example, 27 3 2007</div>
            <div class="govuk-date-input" id="due-date">
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-day">Day</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-day" name="due-date-day" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-month">Month</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-month" name="due-date-month" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-year">Year</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-4"
                        id="due-date-year" name="due-date-year" type="text" inputmode="numeric">
                    </div>
                </div>
            </div>
        </div>
        <div class="govuk-button-group">
            <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
        </div>
    </form>
</main>"
`;

exports[`Appeal Timetables should render "Change statement review Date" page 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-full"><span class="govuk-caption-xl govuk-!-margin-bottom-1">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-!-margin-bottom-7">
        <h1 class="govuk-fieldset__legend--xl govuk-!-margin-top-0">Change statement review due date</h1>
        <div class="govuk-inset-text">The current due date for the statement review is 9 August 2023</div>
    </div>
    <form method="post" novalidate="novalidate">
        <div class="govuk-form-group">
            <div id="due-date-hint" class="govuk-hint">For example, 27 3 2007</div>
            <div class="govuk-date-input" id="due-date">
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-day">Day</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-day" name="due-date-day" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-month">Month</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-month" name="due-date-month" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-year">Year</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-4"
                        id="due-date-year" name="due-date-year" type="text" inputmode="numeric">
                    </div>
                </div>
            </div>
        </div>
        <div class="govuk-button-group">
            <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
        </div>
    </form>
</main>"
`;

exports[`Appeal Timetables should render "Schedule Final Comment Review Date" page 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-full"><span class="govuk-caption-xl govuk-!-margin-bottom-1">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-!-margin-bottom-9">
        <h1 class="govuk-fieldset__legend--xl govuk-!-margin-top-0">Schedule final comment review due date</h1>
    </div>
    <form method="post" novalidate="novalidate">
        <div class="govuk-form-group">
            <div id="due-date-hint" class="govuk-hint">For example, 27 3 2007</div>
            <div class="govuk-date-input" id="due-date">
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-day">Day</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-day" name="due-date-day" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-month">Month</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-month" name="due-date-month" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-year">Year</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-4"
                        id="due-date-year" name="due-date-year" type="text" inputmode="numeric">
                    </div>
                </div>
            </div>
        </div>
        <div class="govuk-button-group">
            <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
        </div>
    </form>
</main>"
`;

exports[`Appeal Timetables should render "Schedule LPA questionnaire Date" page 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-full"><span class="govuk-caption-xl govuk-!-margin-bottom-1">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-!-margin-bottom-9">
        <h1 class="govuk-fieldset__legend--xl govuk-!-margin-top-0">Schedule LPA questionnaire due date</h1>
    </div>
    <form method="post" novalidate="novalidate">
        <div class="govuk-form-group">
            <div id="due-date-hint" class="govuk-hint">For example, 27 3 2007</div>
            <div class="govuk-date-input" id="due-date">
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-day">Day</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-day" name="due-date-day" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-month">Month</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-month" name="due-date-month" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-year">Year</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-4"
                        id="due-date-year" name="due-date-year" type="text" inputmode="numeric">
                    </div>
                </div>
            </div>
        </div>
        <div class="govuk-button-group">
            <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
        </div>
    </form>
</main>"
`;

exports[`Appeal Timetables should render "Schedule issue determination" page 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-full"><span class="govuk-caption-xl govuk-!-margin-bottom-1">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-!-margin-bottom-9">
        <h1 class="govuk-fieldset__legend--xl govuk-!-margin-top-0">Schedule issue determination due date</h1>
    </div>
    <form method="post" novalidate="novalidate">
        <div class="govuk-form-group">
            <div id="due-date-hint" class="govuk-hint">For example, 27 3 2007</div>
            <div class="govuk-date-input" id="due-date">
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-day">Day</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-day" name="due-date-day" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-month">Month</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-month" name="due-date-month" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-year">Year</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-4"
                        id="due-date-year" name="due-date-year" type="text" inputmode="numeric">
                    </div>
                </div>
            </div>
        </div>
        <div class="govuk-button-group">
            <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
        </div>
    </form>
</main>"
`;

exports[`Appeal Timetables should render "Schedule statement review Date" page 1`] = `
"<main class="govuk-main-wrapper " id="main-content" role="main">
    <div class="govuk-grid-row">
        <div class="govuk-grid-column-full"><span class="govuk-caption-xl govuk-!-margin-bottom-1">Appeal 351062</span>
        </div>
    </div>
    <div class="govuk-!-margin-bottom-9">
        <h1 class="govuk-fieldset__legend--xl govuk-!-margin-top-0">Schedule statement review due date</h1>
    </div>
    <form method="post" novalidate="novalidate">
        <div class="govuk-form-group">
            <div id="due-date-hint" class="govuk-hint">For example, 27 3 2007</div>
            <div class="govuk-date-input" id="due-date">
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-day">Day</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-day" name="due-date-day" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-month">Month</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-2"
                        id="due-date-month" name="due-date-month" type="text" inputmode="numeric">
                    </div>
                </div>
                <div class="govuk-date-input__item">
                    <div class="govuk-form-group">
                        <label class="govuk-label govuk-date-input__label" for="due-date-year">Year</label>
                        <input class="govuk-input govuk-date-input__input govuk-input--width-4"
                        id="due-date-year" name="due-date-year" type="text" inputmode="numeric">
                    </div>
                </div>
            </div>
        </div>
        <div class="govuk-button-group">
            <button type="submit" class="govuk-button" data-module="govuk-button">Continue</button>
        </div>
    </form>
</main>"
`;
